api/server/server.go

```diff
@@ -199,6 +198,26 @@ func (s *server) RegisterChain(chainName string, ctx *snow.ConsensusContext, vm
 			)
 		}
 	}
+
+	ctx.Lock.Lock()
+	serviceName, grpcHandler, err := vm.CreateGRPCService(context.TODO())
+	ctx.Lock.Unlock()
+	if err != nil {
+		s.log.Error("failed to create grpc service",
+			zap.String("chainName", chainName),
+			zap.Error(err),
+		)
+		return
+	}
+
+	if serviceName == "" && grpcHandler == nil {
+		return
+	}
+
+	grpcHandler = s.wrapMiddleware(chainName, grpcHandler, ctx)
+	if !s.grpcRouter.Add(ctx.ChainID.String(), serviceName, grpcHandler) {
```

Comment:
I'm assuming we want to force people to use the chain prefix instead of any aliases